packages/core/src/prompts/prompt-registry.ts

```diff
@@ -7,50 +7,38 @@
 import { DiscoveredMCPPrompt } from '../tools/mcp-client.js';
 
 export class PromptRegistry {
-  private prompts: Map<string, DiscoveredMCPPrompt> = new Map();
+  private prompts: DiscoveredMCPPrompt[] = [];
 
   /**
    * Registers a prompt definition.
    * @param prompt - The prompt object containing schema and execution logic.
    */
   registerPrompt(prompt: DiscoveredMCPPrompt): void {
-    if (this.prompts.has(prompt.name)) {
-      const newName = `${prompt.serverName}_${prompt.name}`;
-      console.warn(
-        `Prompt with name "${prompt.name}" is already registered. Renaming to "${newName}".`,
-      );
-      this.prompts.set(newName, { ...prompt, name: newName });
-    } else {
-      this.prompts.set(prompt.name, prompt);
-    }
+    this.prompts.push(prompt);
   }
 
   /**
    * Returns an array of all registered and discovered prompt instances.
+   * Name conflicts are handled by the slash command processor.
    */
   getAllPrompts(): DiscoveredMCPPrompt[] {
-    return Array.from(this.prompts.values()).sort((a, b) =>
-      a.name.localeCompare(b.name),
-    );
+    return [...this.prompts].sort((a, b) => a.name.localeCompare(b.name));
   }
 
   /**
    * Get the definition of a specific prompt.
+   * If multiple prompts share the same name, the first one found is returned.
```

Comment:
This would be a bad idea, but I searched and this isn't being used anywhere, so fine to just remove it!